astacus/node/api.py

```diff
@@ -173,3 +223,38 @@ def cassandra(req: Union[ipc.NodeRequest, ipc.CassandraStartRequest], subop: ipc
 def cassandra_result(*, subop: ipc.CassandraSubOp, op_id: int, n: Node = Depends()):
     op, _ = n.get_op_and_op_info(op_id=op_id, op_name=OpName.cassandra)
     return op.result
+
+
+SnapshotReq = Union[ipc.SnapshotRequestV2, ipc.SnapshotDownloadRequest, ipc.SnapshotClearRequest]
+
+
+def groups_from_snapshot_req(req: SnapshotReq) -> Sequence[SnapshotGroup]:
+    # We merge the list of groups and simple root_globs
+    # to handle backward compatibility if the controller is older than the nodes.
+    groups = [SnapshotGroup(root_glob=root_glob) for root_glob in req.root_globs]
+
+    if isinstance(req, ipc.SnapshotRequestV2):
+        groups += [
+            SnapshotGroup(
+                root_glob=group.root_glob,
+                excluded_names=group.excluded_names,
+                embedded_file_size_max=group.embedded_file_size_max,
+            )
+            for group in req.groups
+            if not any(existing_group.root_glob == group.root_glob for existing_group in groups)
+        ]
+    return groups
+
+
+def snapshotter_from_snapshot_req(req: SnapshotReq, n: Node) -> Snapshotter:
+    groups = groups_from_snapshot_req(req)
+    snapshotter = n.get_or_create_snapshotter(groups)
+    assert snapshotter is not None
```

Comment:
Why do we need this assert ?

Reply:
hmm, indeed seems that we don't; probably repeated the one from inside the op